### PR TITLE
feat: add debug

### LINKED ISSUE #697: Add an option run builds in debug mode
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Description

Add an option to run builds in debug mode.

### Use Case(s)

Provide the option to run a build from `./build.sh` with debug for troubleshooting purposes.

### Potential Configuration

To be determined.

### References

- [x] https://github.com/vmware-samples/packer-examples-for-vsphere/discussions/689

### PR DESCRIPTION
**Pull request by tenthirtyam**
# 

## Summary of Pull Request

- Adds `--help` / `-h` option to `./build.sh` for help.
- Adds `--debug` / `-d` option to `./build.sh` for debug mode.
- Updates `README.md`.

## Type of Pull Request

- [ ] This is a bugfix. `type/bug`
- [x] This is an enhancement or feature. `type/feature` or `type/enhancement`
- [ ] This is a documentation update. `type/docs`
- [ ] This is a refactoring update. `type/refactor`
- [ ] This is a chore. `type/chore`
- [ ] This is something else.
      Please describe:

## Related to Existing Issues

Closes #697

## Test and Documentation Coverage

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

## Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -38,7 +53,12 @@ menu_option_1() {
 
   ### Start the Build. ###
   echo "Starting the build...."
-  packer build -force \
+  if [ "$debug" = true ]
+  then
+    packer build -force -on-error=ask -debug "$INPUT_PATH"
```
Comment (mfvmw) [inline build.sh:58]: Looks like we're double including $INPUT_PATH in all of these build commands now? Once at the start here and then once after (line 67 in this one)

Comment (tenthirtyam) [reply build.sh:58]: 👀 Great catch! Looks like that's always been there and needs to be adjusted!

Comment (tenthirtyam) [reply build.sh]: Resolved in https://github.com/vmware-samples/packer-examples-for-vsphere/commit/992d77ee0693f733be68cbde6d0b6bc454a630a6.

**Review by mfvmw — APPROVED** [review]

**Review by rcroft-work — APPROVED** [review]
I am okay with change.

**Pull request merged**